jupyterhub/app.py

```diff
@@ -3649,12 +3649,20 @@ async def start_service(
                 else:
                     return True
             else:
-                self.log.error(
-                    "Cannot connect to %s service %s at %s. Is it running?",
-                    service.kind,
-                    service_name,
-                    service.url,
-                )
+                if service.managed:
+                    self.log.critical(
+                        "Cannot connect to %s service %s",
+                        service_name,
+                        service.kind,
+                        exc_info=True,
```

Comment:
exc_info=True, not sure if this makes sense here or what it does. I ended up doing a copy paste for this.

> If exc_info does not evaluate as false, it causes exception information to be added to the logging message. If an exception tuple (in the format returned by [sys.exc_info()](https://docs.python.org/3/library/sys.html#sys.exc_info)) or an exception instance is provided, it is used; otherwise, [sys.exc_info()](https://docs.python.org/3/library/sys.html#sys.exc_info) is called to get the exception information.

Reply:
```suggestion
```

I suspect it doesn't make sense here.

Reply:
I think you're right. This is an `else` bock, which means no error. There shouldn't be exc_info to log here.